### PR TITLE
Azure backup add QA tests.

### LINKED ISSUE #15901: Azure Backup Store can be tested on zeebe-io infrastructure

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

In these PR the BackupAcceptence and RestoreAcceptance test were also added to the Azure Backups.
Also some refactor was done in order to remove repeated code among these QA tests.

## Related issues

closes https://github.com/camunda/zeebe/issues/15901

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * Acceptance tests for the backup management API. Tests here should interact with the backups
+ * primarily via the management API, and occasionally assert results on the configured backup store.
+ *
+ * <p>The tests run against a cluster of 2 brokers and 1 gateway, no embedded gateways, two
+ * partitions and replication factor of 1. This allows us to test that requests are correctly fanned
+ * out across the gateway, since each broker is guaranteed to be leader of a partition.
+ *
+ * <p>NOTE: this does not test the consistency of backups, nor that partition leaders correctly
+ * maintain consistency via checkpoint records. Other test suites should be set up for this.
+ */
```
Comment (oleschoenburg) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java:40]: :wrench: At least part of that comment could be on the `BackupAcceptance` test interface

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  @BeforeEach
+  void beforeEach() {
+    final AzureBackupConfig config =
+        new AzureBackupConfig.Builder()
+            .withConnectionString(AZURITE_CONTAINER.getConnectString())
+            .withContainerName(CONTAINER_NAME)
+            .build();
+    store = new AzureBackupStore(config);
+    final BlobServiceClient blobServiceClient = AzureBackupStore.buildClient(config);
+    blobServiceClient.createBlobContainerIfNotExists(CONTAINER_NAME);
```
Comment (oleschoenburg) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java:75]: :question: Isn't that something that the store does for us? Or am I misremembering?

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          final var azure = backup.getAzure();
+
+          backup.setStore(BackupStoreType.AZURE);
+          azure.setBasePath(CONTAINER_NAME);
+          azure.setConnectionString(AZURITE_CONTAINER.getConnectString());
+        });
+  }
+
+  private void configureNode(final TestApplication<?> node) {
+    node.withProperty("management.endpoints.web.exposure.include", "*");
```
Comment (oleschoenburg) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureBackupAcceptanceIT.java:106]: :wrench: I _think_ this isn't necessary anymore, could you give it one test run without this?

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureRestoreAcceptanceIT.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  @BeforeAll
+  static void setupBucket() {
+    final AzureBackupConfig config =
+        new AzureBackupConfig.Builder()
+            .withConnectionString(AZURITE_CONTAINER.getConnectString())
+            .withContainerName(CONTAINER_NAME)
+            .build();
+
+    final BlobServiceClient blobServiceClient = AzureBackupStore.buildClient(config);
+    blobServiceClient.createBlobContainerIfNotExists(CONTAINER_NAME);
```
Comment (oleschoenburg) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/AzureRestoreAcceptanceIT.java:36]: :wrench: Same as before, I _think_ this shouldn't be necessary? Not entirely sure though.

**Review by lenaschoenburg — APPROVED** [review]
Really great job @rodrigo-lourenco-lopes, this looks really good :+1:

**Pull request merged**